### PR TITLE
Major ref

### PR DESCRIPTION
**Pull request by bochkarevko**
Kotlin and Java extractors were merged into one.
Extractor itself is now PsiVisitor, so it is created once and is used for the whole project.
Version of IntelliJ was updated.
Args are now used locally and are not "global".
Intoduced some utilites to make "exiting when something is false/null" cleaner.
A lot of style fixes.

### CODE REVIEW

File: DependencyGraph/src/main/kotlin/org/jetbrains/research/ictl/fileimportance/DependencyExtractor.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import com.intellij.refactoring.suggested.startOffset
+import java.nio.file.Path
+
+class DependencyExtractor(
+    val project: Project,
+    val extension: String,
+    val projectPath: Path
+) : PsiRecursiveElementVisitor() {
+    private val edgesBuffer: MutableList<DependencyEdge> = mutableListOf()
+    private val foundElementOffsets = mutableListOf<Int>()
```
Comment (bochkarevko) [inline DependencyGraph/src/main/kotlin/org/jetbrains/research/ictl/fileimportance/DependencyExtractor.kt]: I am not sure I fully understand why is this required. @vharatian can you explain it please?

Comment (vharatian) [reply DependencyGraph/src/main/kotlin/org/jetbrains/research/ictl/fileimportance/DependencyExtractor.kt]: Do you mean the abstraction? Since the dependency extractors for different languages were different before an abstraction was required for their API.

Comment (bochkarevko) [reply DependencyGraph/src/main/kotlin/org/jetbrains/research/ictl/fileimportance/DependencyExtractor.kt]: Im ean the `foundElementOffsets`. What is it for?
